action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -557,6 +557,27 @@ func (stateDB *StateDBAdapter) RevertToSnapshot(snapshot int) {
 	// restore gas refund
 	if !stateDB.manualCorrectGasRefund {
 		stateDB.refund = stateDB.refundSnapshot[snapshot]
+		delete(stateDB.refundSnapshot, snapshot)
+		for i := snapshot + 1; ; i++ {
+			if _, ok := stateDB.refundSnapshot[i]; ok {
+				delete(stateDB.refundSnapshot, i)
+			} else {
+				break
+			}
+		}
+	}
+	// restore access list
+	stateDB.accessList = nil
+	stateDB.accessList = stateDB.accessListSnapshot[snapshot]
+	{
+		delete(stateDB.accessListSnapshot, snapshot)
+		for i := snapshot + 1; ; i++ {
+			if _, ok := stateDB.accessListSnapshot[i]; ok {
+				delete(stateDB.accessListSnapshot, i)
+			} else {
+				break
+			}
+		}
```

Comment:
reason to move it here, is that L623 may return on "failed on load root", so should clear snapshot before that

and accessList won't be activated before next HF, so it is safe to move it here